server/app/services/export/PdfExporter.java

```diff
@@ -1,46 +1,123 @@
+import java.time.ZoneId;
+import javax.inject.Inject;
+import models.Application;
+import services.applicant.AnswerData;
+import services.applicant.ApplicantService;
+import services.applicant.ReadOnlyApplicantProgramService;
 
 /** PdfExporter is meant to generate PDF files. The functionality is not fully implemented yet. */
 public class PdfExporter {
-  private PDDocument baseDocument;
-  private ImmutableMap<String, String> fieldToValue;
+  private final ApplicantService applicantService;
+  private final Clock clock;
 
-  public PdfExporter(URI documentUrl, Map<String, String> fieldToValue) throws IOException {
-    baseDocument = PDDocument.load(documentUrl.toURL().openStream());
-    this.fieldToValue = ImmutableMap.copyOf(fieldToValue);
+  @Inject
+  PdfExporter(ApplicantService applicantService, Clock clock) {
+    this.applicantService = checkNotNull(applicantService);
+    this.clock = checkNotNull(clock);
   }
-
   /**
-   * Write a PDF containing the filled-in base form to the provided writer. For the PDF to be valid,
-   * the writer should contain no previous writes and should be closed immediately after this call.
-   * This function marshals the output document in memory due to restrictions of the PDDocument
-   * class.
+   * Generates a byte array containing all the values present in the List of AnswerData using
+   * itextPDF.This function creates the output document in memory as a byte[] and is part of the
+   * inMemoryPDF object. The InMemoryPdf object is passed back to the AdminController Class to
+   * generate the required PDF.
    */
-  public void export(Applicant applicant, Writer writer) throws IOException {
-    PDAcroForm form = baseDocument.getDocumentCatalog().getAcroForm();
-    for (Map.Entry<String, String> fToV : fieldToValue.entrySet()) {
-      Optional<String> applicantValue =
-          applicant.getApplicantData().readAsString(Path.create(fToV.getValue()));
-      if (applicantValue.isPresent()) {
-        form.getField(fToV.getKey()).setValue(applicantValue.get());
+  public InMemoryPdf export(Application application) throws DocumentException, IOException {
+    ReadOnlyApplicantProgramService roApplicantService =
+        applicantService
+            .getReadOnlyApplicantProgramService(application)
+            .toCompletableFuture()
+            .join();
+    ImmutableList<AnswerData> answers = roApplicantService.getSummaryData();
+    InMemoryPdf inMemoryPdf = new InMemoryPdf();
+    String applicantNameWithApplicationId =
+        String.format("%s (%d)", application.getApplicantData().getApplicantName(), application.id);
+    String filename =
+        String.format("%s-%s.pdf", applicantNameWithApplicationId, clock.instant().toString());
+    inMemoryPdf.byteArray =
+        buildPDF(
+            answers,
+            applicantNameWithApplicationId,
+            application.getProgram().getProgramDefinition().adminName());
+    inMemoryPdf.fileName = filename;
+    return inMemoryPdf;
```

Comment:
Suggestion (blocking): rather than directly setting the members of `inMemoryPdf`, use a public constructor:
```java
return new InMemoryPdf(fileName, byteArray);
```

In general avoid directly assigning instance variables from outside a class.